m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.RPT.BATCH

```diff
@@ -0,0 +1,751 @@
+ FMHXFERAMOUNT                = MONEY
+ FMHXFERFREQ                  = NUMBER
+ FMHXFERTYPE                  = NUMBER
+ FREQUENCIESCHR               = CHARACTER ARRAY(14)
+ IDTYPESCHR                   = CHARACTER ARRAY(4)
+ NEWLOCFOUND                  = NUMBER
+ TARGETCOMMENT                = CHARACTER
+ TARGETENDDATE                = DATE
+ TARGETSEQUENCE               = NUMBER
+ TARGETSTARTDATE              = DATE
+ TARGETSYMXUSER               = NUMBER
+ TCCOMMENT                    = CHARACTER ARRAY(9)
+ TCCOUNT                      = NUMBER
+ TCCOUNTMAX                   = 9
+ TCID                         = CHARACTER ARRAY(9)
+ TCIDTYPE                     = NUMBER ARRAY(9)
+ TCSTMTDESCR                  = CHARACTER ARRAY(9)
+ TESTACCOUNTS                 = CHARACTER
+ TMPLOOP                      = NUMBER
+ TOACCOUNT                    = CHARACTER
+ TOSLID                       = CHARACTER
+ TOSLIDTYPE                   = CHARACTER
+ TOTRANAMOUNT                 = MONEY
+ TRANSOURCE                   = NUMBER
+ TRANSOURCECHR                = CHARACTER ARRAY(1)
+ TRANSOURCEID                 = CHARACTER(4)
+ XACCTACCESS                  = NUMBER
+ XFERTYPEFOUND                = NUMBER
+ XFERDAY1FOUND                = NUMBER
+ XFERDAY2FOUND                = NUMBER
+END [DEFINE]
+
+SETUP
+[* To run this program against a set of test members ONLY, enter up to 12,
+** 10-digit member numbers, each separated by a comma, in the TESTACCOUNTS
+** variable below. I.E: TESTACCOUNTS="0000012345,0000334567,0000004256"
+*]
+ TESTACCOUNTS=""
+
+ CALL INITIALIZATIONS
+ CALL PROGRAMPROMPTS
+ CALL FETCHLETTERFILEPARAMS
+END [SETUP]
+
+SELECT
+[* If M2M cross-account access is turned on then look through all accounts
+** where the account activity date is on or after the reporting start date
+** otherwise, narrow the focus down to only those accounts with an HB user
+** who has last logged into home banking on or after the reporting start date.
+*]
+ (
+  (
+   TESTACCOUNTS="" OR
+   (
+    TESTACCOUNTS<>"" AND
+    CHARACTERSEARCH(TESTACCOUNTS,ACCOUNT:NUMBER)>0
+   )
+  ) AND
+  (
+   ACCOUNT:CLOSEDATE=DATENULL OR
+   ACCOUNT:CLOSEDATE>TARGETSTARTDATE) AND
+   (
+    (
+     XACCTACCESS=FALSE AND
+     ANY PREFERENCE WITH (PREFERENCE:HBUSERNAME<>"" AND
+                          PREFERENCE:HBLASTLOGONDATE>=TARGETSTARTDATE)
+    ) OR 
+    (
+     XACCTACCESS=TRUE AND
+     ACCOUNT:ACTIVITYDATE>=TARGETSTARTDATE
+    )
+   )
+ )
+END [SELECT]
+
+PRINT TITLE="Banno M2M Transfers Report"
+ CALL MAINREPORTHEADER
+ CALL GETSHARETRANSFERRECS
+ CALL GETTRANSACTIONS
+END [PRINT]
+
+PROCEDURE PROGRAMPROMPTS
+[* Prompts used by this program. The default for each prompt is shown
+** in brackets.
+*]
+ TARGETSYMXUSER=CODEREAD("",
+                         "Enter the target SymXchange user number",
+                         "which must be in the range of 800-899",
+                         "SymXchange User Number [800]")
+
+ TARGETCOMMENT=CHARACTERREAD("",
```

Comment:
I recommend calling NLTS on TARGETCOMMENT right after the character read for if it, if it is not Blank.
Since Some CU's may use the batch prompt instead of the default in the CFG.

Reply:
@jordan-keenan Makes sense.

Reply:
Done